components/brave_wallet_ui/common/hooks/assets-management.ts

```diff
@@ -50,97 +55,41 @@ const findTokensWithMismatchedVisibility = (
   )
 
 export function useAssetManagement() {
-  // selectors
-  const userVisibleTokensInfo = useUnsafeWalletSelector(
-    WalletSelectors.userVisibleTokensInfo
-  )
-  const removedFungibleTokenIds = useUnsafeWalletSelector(
-    WalletSelectors.removedFungibleTokenIds
-  )
-  const removedNonFungibleTokenIds = useUnsafeWalletSelector(
-    WalletSelectors.removedNonFungibleTokenIds
-  )
-  const deletedNonFungibleTokenIds = useUnsafeWalletSelector(
-    WalletSelectors.deletedNonFungibleTokenIds
-  )
-
   // redux
   const dispatch = useDispatch()
 
-  const tokenIsSetAsRemovedInLocalStorage = React.useCallback(
-    (token: BraveWallet.BlockchainToken) => {
-      const assetId = getAssetIdKey(token)
-      return token.isNft || token.isErc1155 || token.isErc721
-        ? removedNonFungibleTokenIds.includes(assetId)
-        : removedFungibleTokenIds.includes(assetId)
-    },
-    [removedNonFungibleTokenIds, removedFungibleTokenIds]
-  )
-
-  const addOrRemoveTokenInLocalStorage = React.useCallback(
-    (token: BraveWallet.BlockchainToken, addOrRemove: 'add' | 'remove') => {
-      if (token.contractAddress === '') {
```

Comment:
What safeguards do we have in this new implementation, that prevents a user from attempting to `delete` a `native` asset?